app/models/manageiq/providers/vmware/cloud_manager/refresh_parser.rb

```diff
@@ -0,0 +1,133 @@
+class ManageIQ::Providers::Vmware::CloudManager::RefreshParser < ManageIQ::Providers::CloudManager::RefreshParser
+  def self.ems_inv_to_hashes(ems, options = nil)
+    new(ems, options).ems_inv_to_hashes
+  end
+
+  def initialize(ems, options = Config::Options.new)
+    @ems        = ems
+    @connection = ems.connect
+    @options    = options || {}
+    @data       = {}
+    @data_index = {}
+    @inv        = Hash.new { |h, k| h[k] = [] }
+  end
+
+  def process_collection(collection, key)
+    @data[key] ||= []
+
+    collection.each do |item|
+      uid, new_result = yield(item)
+      next if uid.nil?
+
+      @data[key] << new_result
+      @data_index.store_path(key, uid, new_result)
+    end
+  end
+
+  def ems_inv_to_hashes
+    log_header = "MIQ(#{self.class.name}.#{__method__}) Collecting data for EMS name: [#{@ems.name}] id: [#{@ems.id}]"
+
+    $log.info("#{log_header}...")
+
+    get_ems
+    get_orgs
+    get_vdcs
+    get_vapps
+    get_vms
+
+    $log.info("#{log_header}...Complete")
+
+    @data
+  end
+
+  private
+
+  def get_ems
+    @ems.api_version = @connection.api_version
+  end
+
+  def get_orgs
+    @inv[:orgs] = @connection.organizations.all.to_a
+  end
+
+  def get_vdcs
+    @inv[:orgs].each do |org|
+      @inv[:vdcs] += org.vdcs.all
```

Comment:
But `<<` is for appending single object and `+=` is for concatenating another array.